app/models/concerns/blacklight/solr/document/marc_export.rb

```diff
@@ -563,5 +550,67 @@ def name_reverse(name)
     temp_name = name.split(", ")
     return temp_name.last + " " + temp_name.first
   end 
+
+  # Convert our Marc citation fields to easily accessible
+  # hash to share functionality across citations
+  # Would be good to use hash.dig('100', 'a') but to_marc.to_hash
+  # creates a pretty ugly hash that wasn't easily dig-ged
+
+  def to_object
+    doc = {}
+    # We'll use the && operator to short circuit if the MARC record
+    # is missing and if not, we'll set it to second value of condition
+    # Some fields are either/or, some are both (if present)
+
+    doc['isbn']       =   record_to_array('020.a')
+    doc['issn']       =   record_to_array('022.b')
+    doc['author']     =   record_to_array('100.a')
+    doc['edition']    =   record_to_array('250.a')
+    doc['scale']      =   record_to_array('255.a')
+    doc['num_pages']  =   record_to_array('300.a')
+    doc['cite_as']   =    record_to_array('524.a')
+    doc['add_entry'] =    record_to_array('700.a')
+    doc['url']       =    record_to_array('856.u')
+   
+    # Publisher, title, series could have multiple fields
+    # So just join the arrays together
+    
+    doc['pub_place'] =    record_to_array('260.a') +
+                          record_to_array('264.a')
+    doc['publisher']  =   record_to_array('260.b') +
+                          record_to_array('264.b')
+    doc['pub_date']   =   record_to_array('260.c') +
+                          record_to_array('264.c')
+    doc['title']      =   record_to_array('245.a') +
+                          record_to_array('245.b')
+    doc['series']     =   record_to_array('440.a') +
+                          record_to_array('490.a')
+
+    doc
+  end
+
+  def record_to_array(marc_field)
+    record_values = []
+    fields = marc_field.split('.')
+    marc_field = fields[0]
+    sub_field = fields[1]
```

Comment:
possible to compress to `marc_field, subfield = marc_field.split('.')`
could rename `marc_field` param  to avoid reuse

Reply:
Yeah I like the idea of putting that onto a single line but not sure what is a good name of the parameter other than "marc_field." Like to be very clear on variable names but you're right it is vulnerable to be overwritten in scope, etc.